pkg/reconciler/taskrun/validate_taskrun.go

```diff
@@ -168,7 +151,7 @@ func findMissingKeys(neededKeys, providedKeys map[string][]string) map[string][]
 
 // ValidateResolvedTask validates task inputs, params and output matches taskrun
 func ValidateResolvedTask(ctx context.Context, params []v1beta1.Param, matrix *v1beta1.Matrix, rtr *resources.ResolvedTask) error {
-	if err := validateParams(ctx, rtr.TaskSpec.Params, params, matrix); err != nil {
+	if err := validateParams(ctx, rtr.TaskSpec.Params, params, matrix.GetAllParams()); err != nil {
```

Comment:
can you please update the docstring for this function?

Reply:
I think this could use a bit more detail: what does it mean for task params and matrix params to "match" the taskrun? (I'm assuming this means all the params declared in the task are present in the taskrun and have the correct types.) Task inputs/outputs also no longer exist.